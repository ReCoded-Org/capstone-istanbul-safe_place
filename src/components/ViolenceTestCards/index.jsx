import React, {useState} from "react";
import { Col, Row } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import "./index.scss";

const ViolenceTestCards = () => {

  const { t } = useTranslation();
  const [translationKey, setTranslationKey] = useState("TestSpec");
  const cardsArr = [...t(`violenceTestPage.${translationKey}`, { returnObjects: true })];
  
  // const TEST_SPEC = [
  //   {
  //     category: "Physical Abuse",
  //     subtitle: "Has your partner:",
  //     checkboxItems: ["Pushed or shoved you"],
  //   },
  //   {
  //     category: "Physical Abuse",
  //     subtitle: "Has your partner:",
  //     checkboxItems: ["Pushed or shoved you"],
  //   },
  //   {
  //     category: "Physical Abuse",
  //     subtitle: "Has your partner:",
  //     checkboxItems: ["Pushed or shoved you"],
  //   },
  // ]; 
  
  const cards = cardsArr.map((test) => (
    <Col className="card">
      <h3>{test.category}</h3>
      <h5>{test.subtitle}</h5>
      <form className="test">
        {test.checkboxItems.map((item) => (
          <div>
            <input type="checkbox" />
            {item}
          </div>
        ))}
      </form>
    </Col>
  ));

  return (
    <Row className="justify-content-md-center">
      {/* <Col className="card">  */}
      {cards}
      {/* </Col> */}
      {/* <Col className="card">
        <h3>Physical Abuse</h3>
        <h5>Has your partner:</h5>
        <form className="test">
          <div>
            <input type="checkbox" />
            Pushed or shoved you?
          </div>
          <div>
            <input type="checkbox" />
            Held you to keep you from leaving?
          </div>
          <div>
            <input type="checkbox" />
            Slapped or bit you?
          </div>
          <div>
            <input type="checkbox" />
            Kicked or choked you?
          </div>
          <div>
            <input type="checkbox" />
            Hit or punched you?
          </div>
          <div>
            <input type="checkbox" />
            Thrown objects at you?
          </div>
        </form>
      </Col> */}
      {/* <Col className="card">
        <h3>Sexual Abuse</h3>
        <h5>Has your partner:</h5>
        <form className="test">
          <div>
            <input type="checkbox" />
            Been jealously angry, assuming you would have sex with others?
          </div>
          <div>
            <input type="checkbox" />
            Insisted you dress in a more sexual/less sexual way than you wanted?
          </div>
          <div>
            <input type="checkbox" />
            Criticized you sexually?
          </div>
          <div>
            <input type="checkbox" />
            Insisted on unwanted or uncomfortable touching?
          </div>
          <div>
            <input type="checkbox" />
            Withheld sex and/or affection?
          </div>
          <div>
            <input type="checkbox" />
            Called you sexual names? (e.g. “whore” or “frigid”)
          </div>
          <div>
            <input type="checkbox" />
            Demanding monogamy from you, while insisting on freedom for self?
          </div>
          <div>
            <input type="checkbox" />
            Forced sex with them or others or forced you to watch others?
          </div>
          <div>
            <input type="checkbox" />
            Forced sex after beating?
          </div>
        </form>
      </Col>
      <Col className="card">
        <h3>Emotional Abuse</h3>
        <h5>Has your partner:</h5>
        <form className="test">
          <div>
            <input type="checkbox" />
            Ignored your feelings?
          </div>
          <div>
            <input type="checkbox" />
            Ridiculed or insulted your most valued beliefs? (i.e. religion,
            heritage, or class)
          </div>
          <div>
            <input type="checkbox" />
            Withheld approval, appreciation or affection as punishment?
          </div>
          <div>
            <input type="checkbox" />
            Continually criticized you, called you names, shouted?
          </div>
          <div>
            <input type="checkbox" />
            Insulted or drove away your friends or family?
          </div>
          <div>
            <input type="checkbox" />
            Humiliated you in private or public?
          </div>
        </form>
      </Col> */}
    </Row>
  );
};

export default ViolenceTestCards;
