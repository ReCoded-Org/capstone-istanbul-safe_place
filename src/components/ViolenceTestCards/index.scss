@import "../../style/variables";

.card {
  background-color: $warningColor;
  border: none;
  border-radius: 1.5rem;
  margin:2rem .5rem 3rem .5rem;
  padding: 1rem ;

  h1 {
    color: $darkColor;
  }

  h5 {
    color: $lightColor;
  }

  .test {
    display: flex;
    flex-direction: column;
    color: $lightColor;
  }

  input {
    margin-right:.5rem;
  }
  
}