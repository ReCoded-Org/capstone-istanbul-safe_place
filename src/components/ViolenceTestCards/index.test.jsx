import React from "react";
import renderer from "react-test-renderer";
import ViolenceTestCards from "./index";

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<ViolenceTestCards t={(key) => key} />).toJSON();
  expect(tree).toMatchSnapshot();
});
