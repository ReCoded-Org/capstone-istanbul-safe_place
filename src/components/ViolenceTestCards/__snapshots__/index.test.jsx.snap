// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="justify-content-md-center row"
>
  <div
    className="card col"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
      <div
        className="col"
      >
        <h3 />
        <h5 />
      </div>
    </div>
    <div
      className="row"
    >
      <form
        className="test col"
      >
        <div>
          <input
            type="checkbox"
          />
          v
        </div>
        <div>
          <input
            type="checkbox"
          />
          i
        </div>
        <div>
          <input
            type="checkbox"
          />
          o
        </div>
        <div>
          <input
            type="checkbox"
          />
          l
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          n
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          T
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          P
        </div>
        <div>
          <input
            type="checkbox"
          />
          a
        </div>
        <div>
          <input
            type="checkbox"
          />
          g
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          S
        </div>
        <div>
          <input
            type="checkbox"
          />
          p
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          0
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          h
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          k
        </div>
        <div>
          <input
            type="checkbox"
          />
          b
        </div>
        <div>
          <input
            type="checkbox"
          />
          o
        </div>
        <div>
          <input
            type="checkbox"
          />
          x
        </div>
        <div>
          <input
            type="checkbox"
          />
          I
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          m
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
      </form>
      <form
        className="test col"
      >
        <div>
          <input
            type="checkbox"
          />
          v
        </div>
        <div>
          <input
            type="checkbox"
          />
          i
        </div>
        <div>
          <input
            type="checkbox"
          />
          o
        </div>
        <div>
          <input
            type="checkbox"
          />
          l
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          n
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          T
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          P
        </div>
        <div>
          <input
            type="checkbox"
          />
          a
        </div>
        <div>
          <input
            type="checkbox"
          />
          g
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          S
        </div>
        <div>
          <input
            type="checkbox"
          />
          p
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          1
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          h
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          k
        </div>
        <div>
          <input
            type="checkbox"
          />
          b
        </div>
        <div>
          <input
            type="checkbox"
          />
          o
        </div>
        <div>
          <input
            type="checkbox"
          />
          x
        </div>
        <div>
          <input
            type="checkbox"
          />
          I
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          m
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
      </form>
      <form
        className="test col"
      >
        <div>
          <input
            type="checkbox"
          />
          v
        </div>
        <div>
          <input
            type="checkbox"
          />
          i
        </div>
        <div>
          <input
            type="checkbox"
          />
          o
        </div>
        <div>
          <input
            type="checkbox"
          />
          l
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          n
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          T
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          P
        </div>
        <div>
          <input
            type="checkbox"
          />
          a
        </div>
        <div>
          <input
            type="checkbox"
          />
          g
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          S
        </div>
        <div>
          <input
            type="checkbox"
          />
          p
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          2
        </div>
        <div>
          <input
            type="checkbox"
          />
          .
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          h
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          c
        </div>
        <div>
          <input
            type="checkbox"
          />
          k
        </div>
        <div>
          <input
            type="checkbox"
          />
          b
        </div>
        <div>
          <input
            type="checkbox"
          />
          o
        </div>
        <div>
          <input
            type="checkbox"
          />
          x
        </div>
        <div>
          <input
            type="checkbox"
          />
          I
        </div>
        <div>
          <input
            type="checkbox"
          />
          t
        </div>
        <div>
          <input
            type="checkbox"
          />
          e
        </div>
        <div>
          <input
            type="checkbox"
          />
          m
        </div>
        <div>
          <input
            type="checkbox"
          />
          s
        </div>
      </form>
    </div>
  </div>
</div>
`;
